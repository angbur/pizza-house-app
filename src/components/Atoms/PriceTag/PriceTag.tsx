import { CSSProperties } from 'react';
import Typography from '../Typography/Typography';

type PriceTagSize = 'lg' | 'md' | 'sm';

type PriceTagProps = {
  price: number;
  size?: PriceTagSize;
};

const PriceTag = ({ price, size = 'lg' }: PriceTagProps) => {
  let scale = 1;

  if (size === 'md') scale = 0.8;
  if (size === 'sm') scale = 0.7;

  const priceTagStyle: CSSProperties = {
    fontSize: `${(scale + 0.05) * 3}rem`,
  };

  return (
    <Typography variant='price-tag' color='primary' style={priceTagStyle}>{`${price}$`}</Typography>
  );
};

export default PriceTag;
