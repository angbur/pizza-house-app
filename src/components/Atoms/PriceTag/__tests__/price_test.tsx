import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import PriceTag from '../PriceTag';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <PriceTag price={12} />
    </ThemeProvider>
  );
};

it('renders correctly price tag', () => {
  const tree = renderer.create(<Wrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
