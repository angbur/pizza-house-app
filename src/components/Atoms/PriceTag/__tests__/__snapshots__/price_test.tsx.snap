// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly price tag 1`] = `
<p
  className="price-tag"
  style={
    Object {
      "color": "#EB853E",
      "fontFamily": "Klepon Ijo",
      "fontSize": "3.1500000000000004rem",
      "letterSpacing": "0",
    }
  }
>
  12$
</p>
`;
